app/ui/src/app/integration/edit-page/step-configure/filter-steps/basic-filter.component.ts

```diff
@@ -4,40 +4,35 @@ import {
   Input,
   Output,
   ViewEncapsulation,
-  OnChanges
+  OnChanges,
+  OnDestroy
 } from '@angular/core';
-import { FormGroup, FormControl, FormArray } from '@angular/forms';
-import {
-  DynamicFormService,
-  DynamicFormControlModel,
-  DynamicFormArrayModel,
-  DynamicInputModel
-} from '@ng-dynamic-forms/core';
+import { FormGroup, FormArray, FormBuilder, Validators } from '@angular/forms';
 
 import { DataShape, IntegrationSupportService } from '@syndesis/ui/platform';
-import { log, getCategory } from '@syndesis/ui/logging';
-import { DATA_MAPPER } from '@syndesis/ui/store';
+import { log } from '@syndesis/ui/logging';
 import {
-  CurrentFlowService,
-  FlowEvent
+  CurrentFlowService
 } from '@syndesis/ui/integration/edit-page';
-import { createBasicFilterModel, findById } from './basic-filter.model';
-import { BasicFilter } from './filter.interface';
+
+import { BasicFilter, getDefaultOps, convertOps, Op, Rule } from './filter.interface';
+import { of, Observable, Subscription } from 'rxjs';
 
 @Component({
   selector: 'syndesis-basic-filter',
   templateUrl: './basic-filter.component.html',
   encapsulation: ViewEncapsulation.None,
   styleUrls: ['./basic-filter.component.scss']
 })
-export class BasicFilterComponent implements OnChanges {
-  basicFilterModel: DynamicFormControlModel[];
-  formGroup: FormGroup;
-  predicateControl: FormControl;
-  predicateModel: DynamicInputModel;
-  rulesArrayControl: FormArray;
-  rulesArrayModel: DynamicFormArrayModel;
+export class BasicFilterComponent implements OnChanges, OnDestroy {
+  form: FormGroup; // our model driven form
+  rulesArray: any = []; // what's a valid type to use here?
```

Comment:
Type here should be an array of these -> https://github.com/syndesisio/syndesis/blob/master/app/ui/src/app/integration/edit-page/step-configure/filter-steps/filter.interface.ts#L8-L12